src/disco/topo/fd_topob.c

```diff
@@ -392,6 +393,95 @@ validate( fd_topo_t const * topo ) {
   }
 }
 
+void
+fd_topob_auto_layout( fd_topo_t * topo ) {
+  /* Incredibly simple automatic layout system for now ... just assign
+     tiles to CPU cores in NUMA sequential order, except for a few tiles
+     which should be floating. */
+
+  char const * FLOATING[] = {
+    "metric",
+    "bencho",
+    "bhole",  // FIREDANCER only
+  };
+
+  char const * ORDERED[] = {
+    "benchg",
+    "benchs",
+    "net",
+    "quic",
+    "verify",
+    "dedup",
+    "pack",
+    "bank",
+    "poh",
+    "pohi",   // FIREDANCER only
+    "shred",
+    "store",
+    "storei", // FIREDANCER only
+    "sign",
+    "gossip", // FIREDANCER only
+    "repair", // FIREDANCER only
+    "replay", // FIREDANCER only
+    "thread", // FIREDANCER only
+    "sender", // FIREDANCER only
+  };
+
+  for( ulong i=0UL; i<topo->tile_cnt; i++ ) {
+    fd_topo_tile_t * tile = &topo->tiles[ i ];
+    tile->cpu_idx = ULONG_MAX;
+  }
+
+  ulong cpu_ordering[ USHORT_MAX ] = { 0UL };
+  ulong num_cpus = fd_numa_cpu_cnt();
+
+  ulong next_cpu_idx = 0UL;
+  for( ulong i=0UL; i<fd_numa_node_cnt(); i++ ) {
```

Comment:
`fd_numa_node_cnt` does syscalls and stuff, so probably cache it in a variable like `num_cpus`